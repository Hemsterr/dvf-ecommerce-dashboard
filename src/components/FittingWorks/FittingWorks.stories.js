// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import FittingWorks from './index'

storiesOf('FittingWorks', module).add('Default', () => (
  <FittingWorks title="I would like to fit myself using Hemster's Fitting Kit" />
))
