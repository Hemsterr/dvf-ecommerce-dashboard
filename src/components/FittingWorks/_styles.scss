.fitting-works {
  display: flex;
  justify-content: space-evenly;
  width: pxToRem(750px);

  &__title {
    padding: pxToRem(35px) 0 pxToRem(39px);
    @include text-size(pxToRem(15px), pxToRem(2px), pxToRem(30px));
  }

  &__btn {
    text-transform: uppercase;
    margin-top: pxToRem(25px);
    width: pxToRem(300px);
  }
}
