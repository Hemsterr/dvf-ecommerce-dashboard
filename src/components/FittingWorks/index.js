// @flow
import React from 'react'

// Components
import FittingItem from '../FittingItem'
import Button from '../Button'

type Props = {
  options: Array<{
    description: string,
    title: string,
  }>,
  title: string,
  handleOnClick: Function,
}

class FittingWorks extends React.PureComponent<Props> {
  static defaultProps = {
    options: [],
    title: '',
    handleOnClick: () => {},
  }

  render() {
    const { options, title, handleOnClick } = this.props
    return (
      <div>
        <p className="fitting-works__title">{title}</p>
        <div className="fitting-works">
          {options.map(item => (
            <FittingItem
              title={item.title}
              description={item.description}
              key={item.title}
            />
          ))}
        </div>
        <Button
          className="btn btn__basic fitting-works__btn"
          label="Select this option"
          type="success"
          size="small"
          handleOnClick={handleOnClick}
        />
      </div>
    )
  }
}

export default FittingWorks
