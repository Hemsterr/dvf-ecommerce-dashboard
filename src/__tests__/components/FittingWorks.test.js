// Libs
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

// Component
import FittingWorks from '../../components/FittingWorks'

const mockProps = {
  title: 'How our Self Service Fitting works',
  options: [
    {
      title: 'fitting',
      description:
        'Our Fit Stylist will meet with you virtually to assist you with fitting your garments',
    },
    {
      title: 'ship',
      description:
        'Ship your garments to us with the prepaid shipping label your Fit Stylist will send via email',
    },
    {
      title: 'deliver',
      description: 'Your finished garments are shipped back to you',
    },
  ],
  handleOnClick: jest.fn(),
}

const wrapper = (props = {}) => {
  const component = shallow(<FittingWorks {...props} />)

  return component
}

describe('FittingWorks component', () => {
  // Snapshot
  test('should render correctly FittingWorks component default props', () => {
    const component = wrapper(mockProps)
    expect(shallowToJson(component)).toMatchSnapshot()
    expect(FittingWorks.defaultProps.title).toEqual('')
    expect(FittingWorks.defaultProps.options).toEqual([])
    expect(FittingWorks.defaultProps.handleOnClick()).toBeUndefined()
  })

  // Testing props
  test('should be rendered with props', () => {
    const wrapperWithProps = wrapper(mockProps)
    expect(wrapperWithProps.find('FittingItem').length).toEqual(
      mockProps.options.length
    )
    expect(
      wrapperWithProps.find('.fitting-works__title').props().children
    ).toEqual(mockProps.title)
  })
})
