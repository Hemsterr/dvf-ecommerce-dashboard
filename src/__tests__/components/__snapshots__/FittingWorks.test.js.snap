// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FittingWorks component should render correctly FittingWorks component default props 1`] = `
<div>
  <p
    className="fitting-works__title"
  >
    How our Self Service Fitting works
  </p>
  <div
    className="fitting-works"
  >
    <FittingItem
      description="Our Fit Stylist will meet with you virtually to assist you with fitting your garments"
      key="fitting"
      title="fitting"
    />
    <FittingItem
      description="Ship your garments to us with the prepaid shipping label your Fit Stylist will send via email"
      key="ship"
      title="ship"
    />
    <FittingItem
      description="Your finished garments are shipped back to you"
      key="deliver"
      title="deliver"
    />
  </div>
  <Button
    className="btn btn__basic fitting-works__btn"
    disabled={false}
    handleOnClick={[MockFunction]}
    label="Select this option"
    size="small"
    type="success"
  />
</div>
`;
